app/models/mixins/retirement_mixin.rb

```diff
@@ -237,14 +236,31 @@ def valid_zone?
     respond_to?(:my_zone) && my_zone.present?
   end
 
-  def setup_event_hash(requester)
-    event_hash = {:retirement_initiator => "system"}
-    event_hash[retirement_base_model_name.underscore.to_sym] = self
-    event_hash[:host] = host if self.respond_to?(:host)
-    if requester
-      event_hash[:userid] = requester
-      event_hash[:retirement_initiator] = "user"
+  def system_context_requester
+    if respond_to?(:evm_owner)
+      evm_owner.id
+    else
+      # todo
```

Comment:
@tinaafitz what's supposed to be here, again?

Reply:
Hint: code.  😁

Reply:
I've some hints.

Reply:
@d-m-u Lets raise an error if we don't have an owner.